pkgs/development/compilers/javacard-devkit/default.nix

```diff
@@ -0,0 +1,86 @@
+{ stdenv, requireFile, unzip, makeWrapper, oraclejdk8 }:
+
+let
+  pkg = "javacard-devkit";
+  version = "2.2.2";
+
+  underscoreVersion = builtins.replaceStrings ["."] ["_"] version;
+
+  downloadUrl = "http://www.oracle.com/technetwork/java/javasebusiness/downloads/java-archive-downloads-javame-419430.html#java_card_kit-2.2.2-oth-JPR";
+in
+
+stdenv.mkDerivation rec {
+  name = "${pkg}-${version}";
+
+  src = requireFile {
+    name = "java_card_kit-${underscoreVersion}-linux.zip";
+    url = downloadUrl;
+    sha256 = "1rzkw8izqq73ifvyp937wnjjc40a40drc4zsm0l1s6jyv3d7agb2";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  unpackPhase = "true";
+
+  buildPhase = ''
+    unzip $src
+
+    cd "java_card_kit-${underscoreVersion}"
+    unzip "java_card_kit-${underscoreVersion}-rr-bin-linux-do.zip"
+
+    chmod +x bin/{apdutool,capdump,capgen,converter,cref,exp2text,jcwde,scriptgen,verifycap,verifyexp,verifyrev}
+    chmod u+w bin/cref
+    patchelf --set-interpreter "$(cat "$NIX_CC/nix-support/dynamic-linker")" bin/cref
+
+    mkdir "$out"
+    cp -R bin "$out"
+    cp -R lib "$out"
+    cp -R api_export_files "$out"
```

Comment:
I'd suggest to put this in `$out/share/javacard-devkit` instead of directly in `$out`.